### ISSUE TITLE
clean up the nav and homepage ui

### ISSUE THREAD

**Issue by cch5ng** [cch5ng/wastenot#26]
after auth is complete

- [x] database

   * lists are associated with a user

   * lists have a type field

- [ ] refactor template lists, routes, test crud ops

   - [x] get template lists (by user, test auth); update logic for user

   - [x] create template list (by user, test auth)

   - [x] delete template list (by user, test auth)
   
   - [x] edit template list (by user, test auth)

- [ ] rethink auth flow (see resc below) 

------

- [ ] include the generic lists (separate the area from the template lists)

   * refactor lists actions (make it more condensed like new thunk pattern)

   * decide if there should be a separate store for shopping list vs template list (or nest them under one store?)

   * add back the routes for lists and test crud ops

- [ ] use a similar design layout to my existing app on surge (maybe rethink colors)

- [x] add nav layout

- [ ] clean up the home page

- [ ] it would be nice to have a feature where it is possible to have a dynamic number of items (like add a section to the list; append to current list length 15) 

- [ ] try to refactor auth (FE action/reducer)

**Comment by cch5ng** [comment]
resc

try out https://reactstrap.github.io/

**Comment by cch5ng** [comment]
120419 > 1216

ListTemplateDetailForm.js (refactor to class, switch back to redux thunk for lists create + BE request)

**Comment by cch5ng** [comment]
454

some issues with timing of when the cookie is getting sent (so right after opening app sometimes the list template retrieval is not happening as expected)

need to follow up with delete; updating redux and figure out what the api should be returning

**Comment by cch5ng** [comment]
1210 > 1137

look at network results

```
TypeError: Cannot read property 'name' of undefined
    at pool.query (\wastenot/server/app/list/table.js:64:43)
    at Query.client.query (/\wastenot/server/node_modules/pg-pool/index.js:334:18)
    at Query.handleReadyForQuery (/\wastenot/server/node_modules/pg/lib/query.js:125:10)
    at Connection.<anonymous> (\astenot/server/node_modules/pg/lib/client.js:244:19)
    at Connection.emit (events.js:198:13)
    at Socket.<anonymous> (wastenot/server/node_modules/pg/lib/connection.js:126:12)
    at Socket.emit (events.js:198:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by cch5ng** [comment]
1211 > 1013

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:3000/api/list/add. (Reason: CORS request did not succeed).
```
